pyproject.toml

```diff
@@ -1,25 +1,111 @@
 [build-system]
 requires = [
-    "setuptools",
-    "wheel",
-    "numpy",
-    "ninja",
-    "pyyaml",
-    "setuptools",
-    "cffi",
-    "typing_extensions",
-    "future",
-    "tensorrt >=8.6,<8.7"
+    "setuptools==68.0.0",
+    "packaging>=23.1",
+    "wheel>=0.40.0",
+    "numpy>=1.25.0",
+    "ninja>=1.11.0",
+    "pyyaml>=6.0",
+    "cffi>=1.15.1",
+    "typing_extensions>=4.7.0",
+    "future>=0.18.3",
+    "tensorrt>=8.6,<8.7",
+    #"torch >=2.0.0,<2.1.0",
+    "torch==2.1.0.dev20230619+cu121",
+    "pybind11==2.6.2"
 ]
+build-backend = "setuptools.build_meta"
 
-# Use legacy backend to import local packages in setup.py
-#build-backend = "setuptools.build_meta:__legacy__"
+[project]
+name = "torch_tensorrt"
+authors = [
+    {name="NVIDIA Corporation", email="narens@nvidia.com"}
+]
+description = "Torch-TensorRT is a package which allows users to automatically compile PyTorch and TorchScript modules to TensorRT while remaining in PyTorch"
+license = {file = "LICENSE"}
+classifiers = [
+    "Development Status :: 5 - Production/Stable",
+    "Environment :: GPU :: NVIDIA CUDA",
+    "License :: OSI Approved :: BSD License",
+    "Intended Audience :: Developers",
+    "Intended Audience :: Science/Research",
+    "Operating System :: POSIX :: Linux",
+    "Programming Language :: C++",
+    "Programming Language :: Python",
+    "Programming Language :: Python :: Implementation :: CPython",
+    "Topic :: Scientific/Engineering",
+    "Topic :: Scientific/Engineering :: Artificial Intelligence",
+    "Topic :: Software Development",
+    "Topic :: Software Development :: Libraries",
+]
+readme = {file = "py/README.md", content-type = "text/markdown"}
+requires-python = ">=3.8"
+keywords = ["pytorch", "torch", "tensorrt", "trt", "ai", "artificial intelligence", "ml", "machine learning", "dl", "deep learning", "compiler", "dynamo", "torchscript", "inference"]
+dependencies = [
+    "torch==2.1.0.dev20230619+cu121",
+    "tensorrt>=8.6,<8.7",
+    "packaging>=23",
+    "numpy>=1.24,<1.25",
```

Comment:
This `numpy` requirement is incompatible with the specification in `requires` above, which has `"numpy>=1.25.0"`. Update for compatibility.

Reply:
See https://github.com/pytorch/TensorRT/pull/2056#discussion_r1251154291, related